soccer/src/soccer/planning/planner_node.cpp

```diff
@@ -27,6 +44,76 @@ PlannerNode::PlannerNode()
     }
 }
 
+rclcpp_action::GoalResponse PlannerNode::handle_goal(const rclcpp_action::GoalUUID& uuid,
+                                                     std::shared_ptr<const RobotMove::Goal> goal) {
+    (void)uuid;
+
+    // TODO(p-nayak): REJECT duplicate goal requests so we aren't constantly replanning them
+
+    // planning::MotionCommand motion_command_ = goal->robot_intent.motion_command;
+
+    SPDLOG_ERROR("robot id {} sent goal", goal->robot_intent.robot_id);
+    return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+}
+
+rclcpp_action::CancelResponse PlannerNode::handle_cancel(
+    const std::shared_ptr<GoalHandleRobotMove> goal_handle) {
+    (void)goal_handle;
+    std::shared_ptr<const RobotMove::Goal> goal = goal_handle->get_goal();
+    SPDLOG_ERROR("action server canceled goal from robot: {}", goal->robot_intent.robot_id);
+    return rclcpp_action::CancelResponse::ACCEPT;
+}
+
+void PlannerNode::handle_accepted(const std::shared_ptr<GoalHandleRobotMove> goal_handle) {
+    // this needs to return quickly to avoid blocking the executor, so spin up a new thread
+    // execute() will block (loop) until completion (either success or canceled by client)
+    using namespace std::placeholders;
+    std::thread{std::bind(&PlannerNode::execute, this, _1), goal_handle}.detach();
+}
+
+void PlannerNode::execute(const std::shared_ptr<GoalHandleRobotMove> goal_handle) {
+    // rate-limit loop to 1ms per iteration
+    rclcpp::Rate loop_rate(1);
+
+    // create ptrs to Goal, Result objects per ActionServer API
+    std::shared_ptr<const RobotMove::Goal> goal = goal_handle->get_goal();
```

Comment:
I'm not sure why it needs to be a shared ptr here, but I'm happy to ignore it for now.